ostd/src/mm/page/meta.rs

```diff
@@ -221,15 +221,49 @@ impl<E: PageTableEntryTrait, C: PagingConstsTrait> PageTablePageMeta<E, C>
 where
     [(); C::NR_LEVELS as usize]:,
 {
-    pub fn new_locked(level: PagingLevel, is_tracked: MapTrackingStatus) -> Self {
+    const WRITE_LOCK_BIT: u32 = 1 << 31;
+
+    pub(crate) fn new_locked(level: PagingLevel, is_tracked: MapTrackingStatus) -> Self {
         Self {
             nr_children: UnsafeCell::new(0),
             level,
             is_tracked,
-            lock: AtomicU8::new(1),
+            lock: AtomicU32::new(Self::WRITE_LOCK_BIT),
             _phantom: PhantomData,
         }
     }
+
+    pub(crate) fn lock_read(&self) {
```

Comment:
Here provides additional implementation logics for `RwLock`. I was thinking that if we could add `acquire_read(write)_lock` and `release_lock` methods to `RwLock` (similar to the `SpinLock`), and make them public in the pt module. Then, the original `RwLock` implementation could be based on these APIs to reduce code duplication and we could directly store an `RwLock<()>` here.

This is not directly related to this PR and we can address it later if you think it is reasonable (to reduce TCB).